### PR TITLE
Guide to use Prometheus collector with Sensu Go

### LINKED ISSUE #850: Guide: Using Prometheus metrics with Sensu Go
Create a new at /guides/prometheus-metrics that shows how to use Prometheus with Sensu

Notes:

- https://github.com/sensu/sensu-prometheus-collector
- instrument your app with a Prometheus library with as little as two lines of code (e.g. https://github.com/portertech/dummy/blob/master/main.go#L10 and https://github.com/portertech/dummy/blob/master/main.go#L40) OR use Prometheus node exporter
- collect the metrics using the Sensu Prometheus Collector (https://github.com/sensu/sensu-prometheus-collector; this is written in Go, but works with Sensu Classic and Sensu Go); this requires installation of one plugin and a single configuration file (on the Sensu backend)
- store the metrics in InfluxDB (recommended), using the built-in Sensu Enterprise (classic) InfluxDB handler, or the new Sensu Go InfluxDB handler (https://github.com/sensu/sensu-influxdb-handler); this requires a single configuration file (for built-in integration on classic; or to prompt Sensu Go to download and install it on Sensu Go)
- graph with Grafana (I'm literally showing the dashboard config in the screenshot; it's very simple)

Left todo on IaC for doc assistance:
- [x] Sensu Go Influx DB handler configuration
- [x] InfluxDB install and configuration
- [x] Grafana install and configuration

### PR DESCRIPTION
**Pull request by rgeniesse**
## Description
Closes #850 

## Motivation and Context
Closes #850 

## Review Instructions
Run through guide in new CentOS 7 image.

### CODE REVIEW

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+- [Install and Configure Prometheus](#install-and-configure-prometheus)
+- [Install and Configure Sensu Go](#install-and-configure-sensu-go)
+- [Install Sensu Prometheus Collector](#install-sensu-prometheus-collector)
+- [Install Sensu InfluxDB Handler](#install-sensu-influxdb-handler)
+- [Install and Configure InfluxDB](#install-and-configure-influxdb)
+- [Install and Configure Grafana](#install-and-configure-grafana)
+
+## What is the Sensu Prometheus Collector?
+
+The [Sensu Prometheus Collector][1] is a sensu check plugin that can collect metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. The allows Sensu to then send metrics to one or more time-series databases, such as InfluxDB or Graphite.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]:
```suggestion
The [Sensu Prometheus Collector][1] is a check plugin that collects metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. This allows Sensu to route the collected metrics to one or more time-series databases, such as InfluxDB or Graphite.
```

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+
+## In this guide
+
+This guide uses CentOS 7 as the operating system with all compoments running on the same compute resource. Commands and steps may change for different distributions or if components are running on different compute resources.
+
+At the end, you will have Prometheus scraping metrics. The Sensu Prometheus Exporter will then query the Prometheus API as a Sensu check, send those to an InfluxDB Sensu handler, which will send metrics to an InfluxDB instance. Finally, Grafana will query InfluxDB to display those collected metrics.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]:
```suggestion
At the end, you will have Prometheus scraping metrics. The Sensu Prometheus Collector will then query the Prometheus API as a Sensu check, send those to an InfluxDB Sensu handler, which will send metrics to an InfluxDB instance. Finally, Grafana will query InfluxDB to display those collected metrics.
```

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+{{< highlight shell >}}
+influx -execute "CREATE DATABASE sensu"
+influx -execute "CREATE USER sensu WITH PASSWORD 'sensu'"
+influx -execute "GRANT ALL ON sensu TO sensu"
+{{< /highlight >}}
+
+### Add Sensu check and handler to complete the pipeline
+
+Using the following resource definition in a file called `resources.yml`:
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]:
```suggestion
Given the following resource definition in a file called `resources.yml`:
```

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+      ]
+    },
+    "timezone": "",
+    "title": "Up or Down Sample 2",
+    "uid": "nb442Vsiz",
+    "version": 1
+  }
+}
+{{< /highlight >}}
+
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]: Feels like this needs a conclusion section to wrap things up.

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+- [Install and Configure Sensu Go](#install-and-configure-sensu-go)
+- [Install Sensu Prometheus Collector](#install-sensu-prometheus-collector)
+- [Install Sensu InfluxDB Handler](#install-sensu-influxdb-handler)
+- [Install and Configure InfluxDB](#install-and-configure-influxdb)
+- [Install and Configure Grafana](#install-and-configure-grafana)
+
+## What is the Sensu Prometheus Collector?
+
+The [Sensu Prometheus Collector][1] is a sensu check plugin that can collect metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. The allows Sensu to then send metrics to one or more time-series databases, such as InfluxDB or Graphite.
+
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]: Feels like we need a section to describe why I would want to do this.

**Review by cwjohnston — CHANGES_REQUESTED** [review]
Looking good! I have a few suggestions that I hope are helpful.

File: content/sensu-go/5.0/guides/prometheus-metrics.md (hunk 4 above)
Comment (rgeniesse) [reply content/sensu-go/5.0/guides/prometheus-metrics.md]: Ah yes, definitely. I'll come up with something.

File: content/sensu-go/5.0/guides/prometheus-metrics.md (hunk 5 above)
Comment (rgeniesse) [reply content/sensu-go/5.0/guides/prometheus-metrics.md]: Agreed. As I attempt to write this though, I am finding myself just reiterating what the Prometheus Collector does vs an actual user use case.

Maybe we can pair up on Monday or during office hours to come up with something that is valuable to readers.

**Comment by rgeniesse** [comment]
@cwjohnston, I've added a Why section. Can you take a look? Still not happy with it, but wanted to get something pushed. Thanks!

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+- [Install and Configure InfluxDB](#install-and-configure-influxdb)
+- [Install and Configure Grafana](#install-and-configure-grafana)
+
+## What is the Sensu Prometheus Collector?
+
+The [Sensu Prometheus Collector][1] is a check plugin that collects metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. This allows Sensu to route the collected metrics to one or more time-series databases, such as InfluxDB or Graphite.
+
+## Why use Sensu with Prometheus?
+
+Prometheus offers an actively maintained [node exportor][5] for reporting hardware and operating system metrics. With those metrics exposed, Sensu can route them to one or more time-series databases, such as InfluxDB or Graphite. Both Sensu and Prometheus can run in parallel, complimenting each other and making use of environments where Prometheus is already deployed.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]:
```suggestion
The Prometheus ecosystem contains a number of actively maintained exporters, such as the [node exporter][5] for reporting hardware and operating system metrics or Google's cAdvisor exporter for monitoring containers. These exporters expose metrics which Sensu can collect and route to one or more time-series databases, such as InfluxDB or Graphite. Both Sensu and Prometheus can run in parallel, complimenting each other and making use of environments where Prometheus is already deployed.  
```

**Review by cwjohnston — DISMISSED** [review]
Some minor wording changes in the prometheus example. We may also want to link to this list of exporters https://prometheus.io/docs/instrumenting/exporters/

File: content/sensu-go/5.0/guides/prometheus-metrics.md (hunk 6 above)
Comment (cwjohnston) [reply content/sensu-go/5.0/guides/prometheus-metrics.md]: Link to https://github.com/google/cadvisor for cAdvisor

**Comment by rgeniesse** [comment]
@cwjohnston good points. I've taken that feedback and added links to both mentioned resources.

File: content/sensu-go/5.0/guides/prometheus-metrics.md [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+      - targets: ['localhost:9090']
+{{< /highlight >}}
+
+Start Prometheus in the background
+
+{{< highlight shell >}}
+nohup .prometheus --config.file=prometheus.yml > prometheus.log 2>&1 &
+{{< /highlight >}}
+
+Ensure Pormetheus is running
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/prometheus-metrics.md]:
```suggestion
Ensure Prometheus is running
```

**Review by apaskulin — APPROVED** [review]
Super clear and polished. I was able to run through this with almost no issues on the first try. Pushed a few updates for style consistency, to move the guide to the getting started section, and to move the dashboard config to `/files`.

:shipit:

**Comment by apaskulin** [comment]
There's been some new versions and content reorganization since this was opened, so I'd suggest waiting to copy to all version in a separate PR.

**Pull request merged**